### DISCUSSION TITLE
NewKernel_d: Get_functior<Kernel, Tag, O, T>

### DISCUSSION THREAD

**Discussion by lrineau** [CGAL/cgal#5897] [General]
@mglisse In `NewKernel_d`, I see 
https://github.com/CGAL/cgal/blob/0175389c108368be27b087207b4549cb48ab5247/NewKernel_d/include/CGAL/NewKernel_d/functor_tags.h#L32-L33

I imagine that the third template parameter, `O`, is an "option", that could be for example `No_lazyness`, if we want a construction from `Epeck` that does not create a graph of construction.

What was your plans for the fourth parameter? I have seen that code:
https://github.com/CGAL/cgal/blob/0175389c108368be27b087207b4549cb48ab5247/NewKernel_d/include/CGAL/NewKernel_d/utils.h#L146-L157
and it seems to be using that fourth parameter (with value `int` or `void`). But I do not see where specializations with `int` are used.

As said in https://github.com/CGAL/cgal/issues/5892#issuecomment-895832199, I would like to review that API `Get_functor`, and then consider adopting it for Kernel_23 as well.

**Comment by mglisse** [comment]
The fourth parameter is only there to help with dispatching / overloading / specializing. IIRC, only the void version is used, but it can be convenient as an implementation detail, when I don't want to provide the specialization, I provide it but with a wrong last parameter (int).

I don't think I ever really used the "option" part of Get_functor, so I cannot promise how well it would work. For instance, I don't seem to forward the option consistently, Cartesian_wrap already drops it, so it wouldn't reach Lazy_cartesian, although that would probably not be too hard to fix.
